be/src/formats/json/nullable_column.cpp

```diff
@@ -205,211 +265,178 @@ static Status add_nullable_native_json_column(Column* column, const TypeDescript
+            offsets->append_numbers(&sz, sizeof(sz));
+            null_column->append(0);
 
-    if (value->is_null()) {
-        nullable_column->append_nulls(1);
-        return Status::OK();
+            return Status::OK();
+        } else {
+            auto err_msg = strings::Substitute("Failed to parse value as array, column=$0", name);
+            return Status::InvalidArgument(err_msg);
+        }
+    } catch (simdjson::simdjson_error& e) {
+        auto err_msg = strings::Substitute("Failed to parse value as array, column=$0, error=$1", name,
+                                           simdjson::error_message(e.error()));
+        return Status::DataQualityError(err_msg);
     }
-
-    RETURN_IF_ERROR(add_map_column(data_column.get(), type_desc, name, value));
-
-    null_column->append(0);
-    return Status::OK();
 }
 
 static Status add_adaptive_nullable_struct_column(Column* column, const TypeDescriptor& type_desc,
                                                   const std::string& name, simdjson::ondemand::value* value) {
+    DCHECK(!value->is_null());
     auto nullable_column = down_cast<AdaptiveNullableColumn*>(column);
-
-    if (value->is_null()) {
-        nullable_column->append_nulls(1);
-        return Status::OK();
-    }
-
     auto& data_column = nullable_column->begin_append_not_default_value();
-
     RETURN_IF_ERROR(add_struct_column(data_column.get(), type_desc, name, value));
-
     nullable_column->finish_append_one_not_default_value();
+    return Status::OK();
+}
 
+static Status add_nullable_struct_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                         simdjson::ondemand::value* value) {
+    DCHECK(!value->is_null());
+    auto nullable_column = down_cast<NullableColumn*>(column);
+    auto& null_column = nullable_column->null_column();
+    auto& data_column = nullable_column->data_column();
+    RETURN_IF_ERROR(add_struct_column(data_column.get(), type_desc, name, value));
+    null_column->append(0);
     return Status::OK();
 }
 
 static Status add_adaptive_nullable_map_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
                                                simdjson::ondemand::value* value) {
+    DCHECK(!value->is_null());
     auto nullable_column = down_cast<AdaptiveNullableColumn*>(column);
-
-    if (value->is_null()) {
-        nullable_column->append_nulls(1);
-        return Status::OK();
-    }
-
     auto& data_column = nullable_column->begin_append_not_default_value();
-
     RETURN_IF_ERROR(add_map_column(data_column.get(), type_desc, name, value));
-
     nullable_column->finish_append_one_not_default_value();
+    return Status::OK();
+}
 
+static Status add_nullable_map_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                                      simdjson::ondemand::value* value) {
+    DCHECK(!value->is_null());
```

Comment:
Why this can't be null?

Reply:
null is processed in `add_nullable_column`
https://github.com/StarRocks/starrocks/blob/main/be/src/formats/json/nullable_column.cpp#L473